### ISSUE TITLE
Batch upsert with AsyncClient fails with Unsupported Media type

### ISSUE THREAD

**Issue by Blobonat** [Orange-OpenSource/python-ngsild-client#7]
Hi,

I tried to make a batch upsert using the AsyncClient:

```
async with AsyncClient(hostname=CONTEXT_BROKER_HOST, port=CONTEXT_BROKER_PORT, tenant=DEFAULT_TENANT) as client:
    await client.batch.upsert(entities, update=True)                
```

But following error log is produced:
```
ngsildclient.api.asyn.client - INFO - Connecting client ...
ngsildclient.api.exceptions - INFO - problemdetails={'type': 'Unsupported Media type', 'title': 'Unsupported Media type', 'detail': 'No content type header provided'}
ngsildclient.api.exceptions - INFO - pd_type='Unsupported Media type'
ngsildclient.api.exceptions - INFO - exception=None
```

With the synchronous client everything works fine.

**Comment by Blobonat** [comment]
I did some investigation and it seems that [this commented line](https://github.com/Orange-OpenSource/python-ngsild-client/blob/904998166f4494e19497a3684186da3d82a589aa/src/ngsildclient/api/asyn/client.py#L144) could be the reason.

**Comment by Blobonat** [comment]
Uncomment this line did work for me. I provided a PR, but of course there could be a reason why the header was initially left out that I missed.

**Comment by fbattello** [comment]
Hi,

You're right the `Content-Type` header is incriminated but uncomment this line may lead to other issues.

This line is a piece of initialization of the default headers, global to the httpx Client.
Meaning that these headers will always be sent.
I prefer not to send `Content-Type` systematically even when HTTP verb is `GET`, that could lead to other issues.

The issue can be solved by setting  `Content-Type` on a case-by-case basis.
For example for batch ops `Content-Type` will be set for CRU but not for D.
And to avoid future confusion I think it's better to remove this comment.

**Comment by Blobonat** [comment]
Hi,

I totally agree with you. I just needed a hot fix for my PoC :)

**Comment by fbattello** [comment]
Fixed in v0.5.1.
